client/app/models/RootStore.ts

```diff
@@ -1,10 +1,52 @@
-import { Instance, SnapshotOut, types } from "mobx-state-tree"
+import { Instance, SnapshotOut, types, flow } from "mobx-state-tree"
+import { CreateUserDto } from "../../../api/src/dtos/users.dto";
+
+const UserModel = types.model("User", {
+  id: types.identifier,
+  email: types.string,
+  password: types.string,
+  cookie: types.optional(types.string, ""),
+});
 
 /**
  * A RootStore model.
  */
-export const RootStoreModel = types.model("RootStore").props({
-})
+export const RootStoreModel = types
+  .model("RootStore")
+  .props({
+    users: types.map(UserModel),
+  })
+  .actions((self) => ({
+    addUser(userSnapshot: any) {
+      const user = UserModel.create(userSnapshot);
+      self.users.put(user);
+    }
+  }))
+  .actions((self) => ({
+    login: flow(function* (email: string, password: string) {
+      const apiUrl = `http://localhost:3000/login`;
+      const user: CreateUserDto = {
+        email: email,
+        password: password,
+      }
+
+      // Call API
+      const response = yield fetch(apiUrl, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(user),
+      });
+
+      if (!response.ok) throw new Error('Invalid email or password please try again.');
+      
+      // Extract data. Create snapshot and addUser to state tree.
+      const data = yield response.json();
+      const userSnapshot = { id: String(data.id), email, password, cookie: data.cookie }
```

Comment:
I've looked into some of the standard conventions for passwords and stuff (per ChatGPT lol), and it says its not a good idea for the API to return the user's password upon successful login. 
- for each subsequent API call that requires authentication, we should be using the cookie/token that we returned
- generally not a good idea to unnecessarily send the password back and forth.

Reply:
Good catch